r/R/py-to-r.R

```diff
@@ -0,0 +1,46 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
To support more Arrow structures here, we possibly need to extend the C-Protocol in future a bit more or would the current protocol suffice for implementing `py_to_r.pyarrow.lib.Table`?

Reply:
I think the current interface would suffice, as Table would have to be "shredded" into RecordBatches and then each exported over

Reply:
Another possibility would be to export each chunk of each table column separately to exactly preserve the chunking structure of the table

Reply:
Shredding seems ok from the API perspective but introduces a lot of unnecessary chunking on most columns. We probably have no consistent way of merging the chunks again on the receiver side.

Exporting each chunk of the columns separately is probably the better thing as it will keep the chunking of the table at the original level but as we don't provide an official API, I would expect similar but slightly varying "proprietary" APIs/implementations to pop up.

Reply:
@xhochy That's a slight risk indeed. We don't really know if this specification will be widely used, and in which ways. Another possible avenue is to distinguish schema and data representation if performance turns out to be critical for some applications (as mentioned in the JNI discussion). If those issues pop up in real use, it will be a motivation for a "C data interface v2".